assets/js/dependencies.js

```diff
@@ -28,13 +29,83 @@ jQuery(function () {
       data: 'dependencyVersion',
       type: 'semver',
     },
+    {
+      data: 'location',
+      createdCell: function (td, _cellData, rowData) {
+        const url = rowData.location
+        try {
+          const parsedUrl = new URL(url)
+          const pathSegments = parsedUrl.pathname.split('/').filter(Boolean)
+          const croppedText = pathSegments.slice(-2).join('/') // final directory + filename
+          $(td).html(`<a href="${url}" target="_blank">${croppedText}</a>`)
+        } catch (e) {
+          $(td).text('')
+        }
+      },
+    },
   ]
 
   createTable({
     id: 'dependenciesTable',
-    ajaxUrl: `/dependencies/data/${dataDependencyType}/${dataDependencyName}`,
+    ajaxUrl: `/dependencies/data/${dataDependencyType}/${dataDependencyName.replace(/\//g, '~')}`,
     orderColumn: 1,
     orderType: 'asc',
     columns,
   })
 })
+
+document.addEventListener('DOMContentLoaded', () => {
+  const form = document.querySelector('form')
+  const typeSelect = document.getElementById('dependencyType')
+  const nameSelect = document.getElementById('dependencyName')
+  const updateButton = document.getElementById('selectDependency')
+  const spinner = document.getElementById('loadingSpinner')
+
+  typeSelect.addEventListener('change', async () => {
+    const selectedType = typeSelect.value
+
+    nameSelect.innerHTML = ''
+    spinner.hidden = false
+    updateButton.disabled = true
+
+    if (!selectedType) {
+      nameSelect.innerHTML = '<option value="">Please select</option>'
```

Comment:
To make it more consistent, would suggest doing this the same way you have further down the page, i.e. lines 82-85